src/cljs/witan/ui/controllers/datastore.cljs

```diff
@@ -437,32 +443,78 @@
                   :kixi.group/id (:kixi.group/id group)
                   :kixi.datastore.metadatastore/sharing-update :kixi.datastore.metadatastore/sharing-conj}))
 
+(defn md-key->update-command-key
+  [k]
+  (keyword (str (namespace k) ".update") (name k)))
+
 (defmethod handle
   :metadata-change
   [event {:keys [kixi.datastore.metadatastore/id] :as md}]
-  (let [current-md (data/get-in-app-state :app/datastore :ds/file-metadata id)
-        md-diff (ldiff md current-md)]
-    (when-not (empty? md-diff)
+  (let [md      (data/get-in-app-state :app/datastore :ds/file-metadata-editing)
+        command (data/get-in-app-state :app/datastore :ds/file-metadata-editing-command)]
+    (when-not (empty? command)
+      ;; is this a good order?
       (set-title! (:kixi.datastore.metadatastore/name md))
       (utils/add-file-flag! id :metadata-saving)
       (save-file-metadata! md)
       (reset-file-edit-metadata! md)
-      (data/command! :kixi.datastore.metadatastore/update "1.0.0"
-                     (merge md-diff {:kixi.datastore.metadatastore/id id}))
+      (reset-file-edit-metadata-command!)
+      (data/command! :kixi.datastore.metadatastore/update "1.0.0" (assoc command :kixi.datastore.metadatastore/id id))
       (data/swap-app-state! :app/datastore update-in [:ds/file-properties id] dissoc :update-errors))))
 
 (defmethod handle
   :reset-edit-metadata
   [event _]
   (let [current (data/get-in-app-state :app/datastore :ds/current)
         md (data/get-in-app-state :app/datastore :ds/file-metadata current)]
-    (reset-file-edit-metadata! md)))
+    (reset-file-edit-metadata! md)
+    (reset-file-edit-metadata-command!)))
+
+(defn kw-op->md-op
+  [operation]
+  (get {:assoc :set
+        :update-conj :conj
+        :update-disj :disj}
```

Comment:
Seems odd that these are different. Why not just use :set :conj :disj everywhere?